src/miner.cpp

```diff
@@ -351,8 +296,10 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn)
         pblocktemplate->vTxSigOps[0] = GetLegacySigOpCount(pblock->vtx[0]);
 
         CValidationState state;
-        if (!TestBlockValidity(state, *pblock, pindexPrev, false, false))
-            throw std::runtime_error("CreateNewBlock(): TestBlockValidity failed");
+        if (!TestBlockValidity(state, *pblock, pindexPrev, false, false)) {
+            LogPrintf("ERROR: CNB TestBlockValidity failed: %s\n",FormatStateMessage(state));
+            return NULL;
```

Comment:
Exceptions here should not crash the node... it just gets returned as a nicer JSON-RPC error.